packages/svelte/tests/validator/samples/inline-new-class-2/input.svelte

```diff
@@ -0,0 +1,6 @@
+<script>
+	class Foo {
+		foo = $state(0)
+	}
+	const a = new Foo();
```

Comment:
Warning in this case would be overly persnickety — we don't want to force people to do this:

```svelte
<script context="module">
  class Foo {
    foo = $state(0)
  }
</script>

<script>
  const a = new Foo();
</script>
```

As I see it, the goal here is to discourage people from doing this sort of thing:

```js
function blah() {
  let x = $state(...);

  class Foo {
    whatever() {
      return x;
    }
  }

  return new Foo();
}
```

For classes that _don't_ have methods that close over local stuff, Svelte should be hoisting the definition for us (another reason we need a general-purpose hoisting strategy).

Reply:
Addressed in latest commit.